### DISCUSSION TITLE
Microsoft vcpkg ecosystem optimization for "windows on arm"

### DISCUSSION THREAD

**Discussion by Robinsci** [microsoft/vcpkg#33290] [General]
Hi all,

Vcpkg has supported for more than 2,000 C++ packages and libraries. But it is found that some packages or libraries are not supported for "windows on arm". I have filtered all of them and get a result: the number of packages which support x86/x64 but not "windows on arm" reaches more than 200, such as tensorflow, opencv2, boost-coroutine, cudnn, libebur128 and so on. 

Does Microsoft have a plan to advance this issue? Maybe support more packages on "windows on arm" which are already supported on x86/x64.

I'm looking forward to your reply!

**Comment by autoantwort** [comment]
I think this is mostly a upstream issue. If tensorflow, opencv2, boost-coroutine, cudnn, libebur128 do not supports windows on arm initially vcpkg cannot do anything about it.

You can use `--allow-unsupported` to build the packages on windows on arm.

**Comment by walbourn** [comment]
This is indeed largely an 'upstream' issue which means driving the request back to the original repositories.

Note that *Windows on ARM64* is the main ask here, as ARM32 is deprecated. See [Microsoft Learn](https://learn.microsoft.com/en-us/windows/arm/arm32-to-arm64).

**Comment by Osyotr** [comment]
Hmm doesn't arm64 windows support x64 binaries? Would it make sense to build x64 binaries if arm is not supported by upstream (with a proper warning)?

**Reply by autoantwort** [reply]
I don't think that you can link x64 and arm64 libs at the same time, you have to use [Arm64EC](https://learn.microsoft.com/en-us/windows/arm/arm64ec), but you can run a x64 executable on arm64.
